setup/src/setup-haskell.ts

```diff
@@ -36,17 +36,16 @@ export default async function run(
 
     if (opts.cabal.enable)
       await core.group('Setting up cabal', async () => {
-        await exec('cabal', ['user-config', 'update'], {silent: true});
-        const configFile = await cabalConfig();
-
         if (process.platform === 'win32') {
+          await exec('cabal', ['user-config', 'update'], {silent: true});
+          const configFile = await cabalConfig();
           fs.appendFileSync(configFile, 'store-dir: C:\\sr\n');
           core.setOutput('cabal-store', 'C:\\sr');
+          await exec('cabal user-config update');
```

Comment:
Is there a reason that this now does user-config update twice on windows and not at all on linux/macOS?

Reply:
The original code called `user-config update` twice on all platforms, both Windows and non-Windows. As far as I can tell, this was a no-op on non-Windows platforms, so that's why I moved them into the `if` clause.

On Windows, I believe the first `user-config update` was to ensure the config file exists (it would create a default config file if it didn't), while the second one would reformat the config file after the editing (`fs.appendFileSync`).

Reply:
That's the reasoning behind the two user-config updates on windows, yeah. Interesting that it's seemingly a no-op on non windows platforms though.

Looking through some of the documentation, I see that `cabal update` will create the config file if it doesn't exist (for windows I "need" to mess with it before that happens cause reasons). So for non windows, it is essentially a no-op.

However, there is one situation in which it is not a no-op; `cabal user-config update` automatically migrates configuration files made by older versions of cabal, so it can make sense to call it if you had a system that ran setup-haskell with an old version of cabal and then ran setup-haskell with a new version of cabal (in the same job). A bit of a long shot, admittedly, but technically possible.

That said, it's enough of a corner case that I'm fine with it not happening in the action; putting `cabal update` in there by default was likely a mistake anyway, so the less we exec out to things the better.